packages/auth/src/grant/routes.ts

```diff
@@ -253,3 +255,76 @@ async function continueGrant(
     }
   }
 }
+
+// eslint-disable @typescript-eslint/no-explicit-any
+function isAccessTokenUpdateParameter(
+  param: any
+): param is {
+  access: AccessRequest[]
+} {
+  return (
+    param &&
+    param.access &&
+    Array.isArray(param.access) &&
+    param.access.length > 0 &&
+    param.access.every((a: any) => isAccessRequest(a))
+  )
+}
+// eslint-enable
+
+async function updateGrant(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  // TODO: httpsig validation
+  const { continueId } = ctx.params
+  const continueToken = (ctx.headers['authorization'] as string)?.split(
+    'GNAP '
+  )[1]
+  const {
+    interact_ref: interactRef,
+    access_token: accessToken
+  } = ctx.request.body
+
+  if (
+    !continueId ||
+    !continueToken ||
+    !interactRef ||
+    !isAccessTokenUpdateParameter(accessToken)
+  ) {
+    ctx.status = 401
+    ctx.body = {
+      error: 'invalid_request'
+    }
+  } else {
+    const { config, accessTokenService, grantService, accessService } = deps
+    const grant = await grantService.getByContinue(
+      continueId,
+      continueToken,
+      interactRef
+    )
+
+    if (grant) {
+      await grantService.update(grant.id, accessToken.access)
+      const updatedAccessToken = await accessTokenService.create(grant.id)
```

Comment:
I think we should only be returning an `access_token` if one was already issued, rather than creating one since modifying the grant request shouldn't be a way to authorize it

Reply:
`access_token` doesn't seem required, and it seems like it should be issued if the grant was modified in a way where the previous access already overlaps with the new one (like if the amount of access was reduced in the modification). Otherwise my personal thought is that if _more_ access was added in the grant update, we should expect the RO to authorize it down the line and return the grant to `pending`.

```
When the AS receives a valid modification request, the AS MUST place
   the grant request into the _processing_ state and re-evaluate the
   authorization in the new context created by the update request.

The client instance MAY include the interact field as described in
   [Section 2.5](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-2.5).  Inclusion of this field indicates that the client
   instance is capable of driving interaction with the end user, and
   this field replaces any values from a previous request.  The AS MAY
   respond to any of the interaction responses as described in
   [Section 3.3](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-3.3), just like it would to a new request.
```